index.js

```diff
@@ -3,26 +3,50 @@
 var path = require('path');
 var extend = require('extend-shallow');
 var isNegated = require('is-negated-glob');
+var isAbsolute = require('is-absolute');
 
 module.exports = function(glob, options) {
+  // shallow clone options
   var opts = extend({}, options);
-  opts.cwd = opts.cwd ? path.resolve(opts.cwd) : process.cwd();
+
+  // ensure cwd is absolute
+  var cwd = path.resolve(opts.cwd ? opts.cwd : process.cwd());
+  if (process.platform === 'win32' || opts.unixify === true) {
+    cwd = unixify(cwd);
+  }
+
+  var rootDir = opts.root;
+  // if `options.root` is defined, ensure it's absolute
+  if (rootDir && !isAbsolute(opts.root)) {
+    if (process.platform === 'win32' || opts.unixify === true) {
+      rootDir = unixify(rootDir);
+    }
+    rootDir = path.resolve(rootDir);
+  }
 
   // store last character before glob is modified
   var suffix = glob.slice(-1);
 
+  // check to see if glob is negated (and not a leading negated-extglob)
   var ing = isNegated(glob);
   glob = ing.pattern;
 
-  if (opts.root && glob.charAt(0) === '/') {
-    glob = path.join(path.resolve(opts.root), '.' + glob);
-  } else {
-    glob = path.resolve(opts.cwd, glob);
+  // make glob absolute
+  if (rootDir && glob.charAt(0) === '/') {
+    glob = path.join(rootDir, glob);
+  } else if (!isAbsolute(glob)) {
+    glob = path.resolve(cwd, glob);
   }
 
+  // if glob had a trailing `/`, re-add it now in case it was removed
   if (suffix === '/' && glob.slice(-1) !== '/') {
     glob += '/';
   }
 
+  // re-add leading `!` if it was removed
   return ing.negated ? '!' + glob : glob;
```

Comment:
I agree with you but you are using `path.resolve` on line 38 and `path.join` on line 36 which are both adding Windows path separators into the glob, which is the incorrect behavior.  This library shouldn't be replacing escape sequences users put in their globs, but it shouldn't be adding those escape sequences itself.  Does that make sense?